internal/usecases/houses.go

```diff
@@ -0,0 +1,109 @@
+package usecases
+
+import (
+	"context"
+	"imobiliaria/internal/models"
+	"imobiliaria/internal/usecases/errors"
+)
+
+type Houses interface {
+	GetHouse(ctx context.Context, id int) (*models.House, error)
+	CreateHouse(ctx context.Context, house *models.House) (*models.House, error)
+	GetHouses(ctx context.Context) ([]*models.House, error)
+	UpdateHouse(ctx context.Context, house *models.House) (*models.House, error)
+	DeleteHouse(ctx context.Context, id int) error
+
+	GetHousesByUserID(ctx context.Context, id int) ([]*models.House, error)
+}
+
+func (u *usecases) GetHouse(ctx context.Context, id int) (*models.House, error) {
+	if id <= 0 {
+		return nil, errors.NewError(
+			"id should be defined",
+			errors.ErrorCodeInvalid,
+			nil,
+		)
+	}
+
+	// HTTP 500 é uma escolha de design
+	return u.repo.GetHouse(ctx, id)
```

Comment:
To avoid expose an error from the database to the end user, you could add a custom error when the database operation fails.